Content.Client/UserInterface/Systems/Ghost/Controls/Roles/GhostRolesEui.cs

```diff
@@ -77,35 +78,59 @@ public override void HandleState(EuiStateBase state)
 
             if (state is not GhostRolesEuiState ghostState)
                 return;
+
+            // We must save BodyVisible state, so all Collapsible boxes will not close
+            // on adding new ghost role.
+            // Save the current state of each Collapsible box being visible or not
+            var collapseStates = new Dictionary<int, bool>();
+            foreach (var collapsible in _window.GetAllCollapsibleBoxes())
+            {
+                if (int.TryParse(collapsible.Name, out var collapsibleId))
+                {
+                    collapseStates[collapsibleId] = collapsible.BodyVisible;
+                }
+            }
```

Comment:
None of this logic should be in `GhostRolesEui`, and especially it should not be reading control state. Let `GhostRolesWindow` store this state.